packages/devtools_app/lib/src/memory/flutter/memory_controller.dart

```diff
@@ -491,12 +614,62 @@ class MPChartData {
   /// Datapoint entries for each external memory value.
   final List<Entry> externalHeap = <Entry>[];
 
+  /// Prune entries plotted based on number of sample
+  /// minutes to display.
+  void prune(int minutesToDisplay) {
+    assert(minutesToDisplay > 0);
+
+    final len = used.length;
+
+    // Entries should match.
+    // All entries lengths are the same.
+    assert(len == capacity.length && len == externalHeap.length);
+    // All entries starting timestamps match,
+    assert(used.isNotEmpty &&
+        used[0].x == capacity[0].x &&
+        capacity[0].x == externalHeap[0].x);
+    // All entries ending timestamps match.
+    assert(used.isNotEmpty &&
+        used[len - 1].x == capacity[len - 1].x &&
+        capacity[len - 1].x == externalHeap[len - 1].x);
+
+    // Compute a new starting index from length - N minutes.
+    final timeLastSample = used.last.x;
+    var index = len - 1;
+    for (; index > 0; index--) {
```

Comment:
this syntax feels strange. Why not just `for (var index = len - 1; index > 0; index--)`

Reply:
Could do this:
    var index ;
    for (index = len - 1; index > 0; index--) {
..}

Need access to index outside of for loop.  This seems reasonable to me.